contracts/test/walletAction-test.ts

```diff
@@ -176,6 +180,56 @@ describe("WalletActions", async function () {
     );
   });
 
+  it("should send ETH with function call via fallback expander", async function () {
+    // send money to sender bls wallet
+    const sendWallet = await fx.lazyBlsWallets[0]();
+    const ethToTransfer = parseEther("0.001");
+    await fx.signers[0].sendTransaction({
+      to: sendWallet.address,
+      value: ethToTransfer,
+    });
+
+    const MockAuction = await ethers.getContractFactory("MockAuction");
+    const mockAuction = await MockAuction.deploy();
+    await mockAuction.deployed();
+
+    expect(await fx.provider.getBalance(sendWallet.address)).to.equal(
+      ethToTransfer,
+    );
+    expect(await fx.provider.getBalance(mockAuction.address)).to.equal(0);
+
+    const bundle = await sendWallet.signWithGasEstimate({
+      nonce: BigNumber.from(1),
+      actions: [
+        {
+          ethValue: ethToTransfer,
+          contractAddress: mockAuction.address,
+          encodedFunction: mockAuction.interface.encodeFunctionData("buyItNow"),
+        },
+      ],
+    });
+
+    await (
+      await fx.blsExpanderDelegator.run(
+        bundleCompressedOperations(
+          [
+            compressAsFallback(
+              Fixture.expanderIndexes.fallback,
+              bundle.senderPublicKeys[0],
+              bundle.operations[0],
+            ),
+          ],
+          bundle.signature,
+        ),
+      )
+    ).wait();
```

Comment:
#nitpick I dislike nested awaits as it can be difficult to:
 - Understand what the multi async calls are just by looking at code
 - Can make debugging more difficult. On a throw, which await scope was it?

```suggestion
    const txn = await fx.blsExpanderDelegator.run(
        bundleCompressedOperations(
          [
            compressAsFallback(
              Fixture.expanderIndexes.fallback,
              bundle.senderPublicKeys[0],
              bundle.operations[0],
            ),
          ],
          bundle.signature,
        ),
      )
    );
    await txn.wait();
```

Reply:
Hmm, I have almost the opposite opinion - that nested awaits (when used sparingly) are helpful in keeping the code concise.

> Can make debugging more difficult. On a throw, which await scope was it?

I'm not sure I understand. On a throw, the stack trace should point to the particular call. The stack doesn't merely point to the offending statement, instead it points to the specific location inside the statement. (Also, this point would apply equally to all nested function calls, not just awaits, right?)

Having said all that, I think it's fair this statement is a bit bloated. Extracting the `.run` call leaves us with a statement that still mostly has this bloat though. I'd much rather extract the compressed bundle:

```ts
const compressedBundle = bundleCompressedOperations(
  [
    compressAsFallback(
      Fixture.expanderIndexes.fallback,
      bundle.senderPublicKeys[0],
      bundle.operations[0],
    ),
  ],
  bundle.signature,
);

await (
  await fx.blsExpanderDelegator.run(compressedOperation)
).wait();
```

Reply:
> I'm not sure I understand. On a throw, the stack trace should point to the particular call. The stack doesn't merely point to the offending statement, instead it points to the specific location inside the statement.

Yes, but I would argue that it is easier to find the statement that failed when not nested. Example:

`nestedAwait.js`
```javascript
const txnFactory = async (fail) => ({
  wait: async () => {
    if (fail) {
      throw new Error("txn failed");
    } 
  },
});

(async() => {
  try {
    await (await txnFactory(true)).wait();
  } catch (err) {
    console.error(err);
  }

  console.log("---");

  try {
    const txn = await txnFactory(true);
    await txn.wait();
  } catch (err) {
    console.error(err);
  }

})();
```
```sh
node nestedAwait.js 
Error: txn failed
    at Object.wait (/home/jacque006/workspaces/bls-wallet/nestedAwait.js:4:13)
    at /home/jacque006/workspaces/bls-wallet/nestedAwait.js:11:36
---
Error: txn failed
    at Object.wait (/home/jacque006/workspaces/bls-wallet/nestedAwait.js:4:13)
    at /home/jacque006/workspaces/bls-wallet/nestedAwait.js:20:15
```
For the nested await, I need to find the *character column* to find the statement that threw.
When not nested, I know the await on line 20 failed. IMO much easier to find. It also gets worse if the code your debugging is transpiled.

> (Also, this point would apply equally to all nested function calls, not just awaits, right?)

Yes, before NodeJS went more promisy it was referred to as Callback Hell :)

Reply:
> Yes, before NodeJS went more promisy it was referred to as Callback Hell :)

Ah, the good old days /s

Reply:
@jacque006 Point taken about the improved stack trace. Appreciate the isolated demo code 🚀.

I do feel it is not enough though since the same applies to all nested function calls, and the verbosity of flattening all function calls would not be worth it.

How do you feel about `await receiptOf(contract.fn(args))`?